### ISSUE TITLE
DataTable column resizer control does not provide enough affordance

### ISSUE THREAD

**Issue by halocline** [grommet/grommet#7273]
### Is this something you or your team is interested in contributing a pull request towards?

Yes.

### Expected Behavior

Resizer control should provide [adequate click target size](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html) and not interfere with the cells contents.

### Actual Behavior

DataTable column resizer control does not provide enough affordance between itself and the header cell content when columns are right-aligned. This is especially true when consuming `grommet-theme-hpe`. 

Reported in [this issue](https://hpe.atlassian.net/browse/GLCP-181403?focusedCommentId=1094729).

### URL, screen shot, or Codepen exhibiting the issue

[Storybook](https://storybook.grommet.io/?path=/story/visualizations-datatable-resizable-columns--resizable-data-table&globals=theme:hpe)

![Screenshot 2024-07-16 at 1 36 08 PM](https://github.com/user-attachments/assets/0b4edde9-c212-401f-89f7-0cc15d451396)

### Steps to Reproduce

1. View [Storybook](https://storybook.grommet.io/?path=/story/visualizations-datatable-resizable-columns--resizable-data-table&globals=theme:hpe)

### Your Environment

- Grommet version: 2.37.0
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by jcfilben** [comment]
I think there are two accessibility issues here, the click target size is not large enough and the resizer is not keyboard accessible. 

I looked at some of the other design systems from this [list](https://designsystems.surf/components/data-table) to see how they are handling column resizing in DataTables. The majority of the ones I looked at do not offer column resizing as an option. There were two that support column resizing:
- https://ag-grid.com/react-data-grid/column-sizing/
- https://www.lightningdesignsystem.com/components/data-tables/

The salesforce column resizer doesn't seem to be working. The ag-grid column resizer looks nice visually but doesn't meet the click target requirements or support keyboard interaction.

### ag-grid column resizer

The ag-grid column resizer is nicely positioned directly between the columns with a little bit of click target overlap on either side. To achieve the visual alignment of the resizer control ag-grid is using negative pixel values, something I'm thinking we should avoid if possible.

<img width="513" alt="Screenshot 2024-08-02 at 3 56 24 PM" src="https://github.com/user-attachments/assets/7360c885-8cbc-4177-a136-8fd77036df1d">

<img width="986" alt="Screenshot 2024-08-02 at 3 56 48 PM" src="https://github.com/user-attachments/assets/bd82eb95-b3fb-4b98-9f0e-d350e6de7690">

### Keyboard accessibility approach

I found this [article](https://dev.to/madsstoumann/accessible-resizable-table-columns-5eof) and corresponding [codepen](https://codepen.io/stoumann/pen/poNjMJV) which show a clever approach to make column resizers keyboard accessible by utilizing range input. I think there are some issue with this particularly with how screenreaders would try to read this out but it is a nice starting point.

<img width="979" alt="Screenshot 2024-08-02 at 3 57 36 PM" src="https://github.com/user-attachments/assets/feb6379c-3548-4643-81bc-26430b3ce4d1">

<img width="973" alt="Screenshot 2024-08-02 at 3 57 50 PM" src="https://github.com/user-attachments/assets/1a76de90-1de2-4660-bb7a-a64dc640c648">

### Summary

I think both of the issues are possible to resolve but involve tricky code and will be time consuming. It seems like column resizing isn't supported in a lot of the design systems. I'm wondering if our Data + DataTableColumns component solves a similar use case to resizing and maybe is a better solution than resizing columns because it allows the user to control what columns they are seeing, the order of columns, and their visibility. Overall I am just wondering how much we want to invest in the DataTable column resizer.

**Comment by jcfilben** [comment]
Started exploring fixing the DataTable resizer. Code so far is in this branch: https://github.com/jcfilben/grommet/tree/datatable-resizer. There are still lots of issues with the resizer, this code is definitely still a work in progress.

**Comment by britt6612** [comment]
This also does not pass the following as far as WCAG rules:
2.1.1 Keyboard - no keyboard support for datatable column resize
2.1.3 Keyboard (no exception) - no keyboard support for datatable column resize
2.5.5 TargetSize (enhanced) - Don't think resizable cols meets target size
2.5.7 Dragging Movements - column resize can't be done via keyboard
2.5.8 (Target Size (Minimum)) - column resize not large enough

**Comment by jcfilben** [comment]
Issue resolved and wcag documentation updated
